src/mocks/mock-request.js

```diff
@@ -80,7 +82,16 @@ MockRequest.prototype = {
       ? this.url(request.url(), request.params())
       : sortedUrl(this.url) === sortedUrl(request.url())
 
-    return this.method === request.method() && urlMatch && bodyMatch()
+    // If the mock has no headers specified, don't use it for matching
```

Comment:
I don't think we should match on unspecified headers. It's just really unpractical in case you have some global middleware that's adding request identifiers or some timestamps or something to have to make those match in your mock configuration. I would prefer if we match only those headers that are actually specified in the mock config.

Reply:
Hey Nevon 👋 
Thanks for taking the time to review the PR :)~

I'm not sure what you mean by this:
>  I would prefer if we match only those headers that are actually specified in the mock config.

Are you suggesting that we should ignore additional headers? e.g. a mock that defines a header `a` matches against a request that defines `a` and `b`, but won't match against one that only defines `b`.

If that's your suggestion, I can make it happen :)~

Reply:
Exactly, that's what I mean. It makes the header validation inconsistent with the validation for query parameters and so on, which is a drawback, but I just see the difficulty of mocking some values set dynamically by global middleware unrelated to what you're testing.

Reply:
Has been addressed in the latest version 🎉